### PR TITLE
refactor debian packaging split binaries, libraries, headers

### PR DESCRIPTION
**Pull request by rafael2k**
This work aims to aid Debian-based Linux distributions to re-include WiringPi to repository.

### CODE REVIEW

**Comment by mwallner** [comment]
I've retargeted the PR towards the 'ubuntu' branch, in which we'll prepare the packaging of the current 3-way implementation of debian packages. 

we're aiming for
* libwiringpi3 
* wiringpi
* libwiringpi3-dev

**Comment by mwallner** [comment]
we do not want to change the default Rapsbian / PI-OS debian packages, I'm going to rebase/coauthor your changes onto the newly introduced ubuntu packages. going forward, we're going to put out two flavors: 
* the one that everyone on debian/raspbian already known
* the 3-split ubuntu packages

**Comment by rafael2k** [comment]
Cool! For "external" deb package distribution, I agree it is easier for the user to just download one .deb. But for acceptance by Debian and/or Ubuntu, minimal guidelines need to be followed (eg. separating apps, libraries and headers in different packages), this is why I think it is nice nice for us (to help reduce the burden of distro maintainers) to keep a debian directory in a more or sane state for them to start with (and for me to create packages for my debian reposity with WiringPi packages : ) http://packages.hermes.radio/  )

**Comment by mwallner** [comment]
I see, that makes sense, in this way, how about doing a 4-packages split?

* wiringpi => 'the old one', everything batteries included
* wiringpi-bin => just the binaries / cli tools / manpages
* libwiringpi3 => the development libraries
* libwiringpi3-dev => the development sources

CC @mstroh76

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,49 @@
+wiringpi (3.1-1) unstable; urgency=low
```
Comment (mwallner) [inline debian/changelog:1]: we do not consider published releases to be unstable :wink:

Comment (mwallner) [reply debian/changelog:1]: also, the changelog will most likely need to be autogenerated going forward.

File: debian/control [hunk 2]
```diff
@@ -1,29 +1,38 @@
 Source: wiringpi
-Section: electronics
 Priority: optional
-Maintainer: Ian Jackson <ijackson@chiark.greenend.org.uk>
-Standards-Version: 3.8.0
-Homepage: https://github.com/WiringPi/WiringPi
-Build-Depends: debhelper (>= 8)
-
-Package: libwiringpi2
+Maintainer: Rafael Diniz <rafael@rhizomatica.org>
```
Comment (mwallner) [inline debian/control:3]: The WiringPi Community <wiringpi@gc2.at>

File: debian/copyright
Comment (mwallner) [inline debian/copyright:1]: we'll need to add "The WiringPi Community" with a link to the GitHub Org in this file - LGPL-3.0+, starting 2024

File: debian/changelog [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+
+  * New maintainers. Support for Pi 5 and many improvements.
+
+ -- Rafael Diniz <rafael@rhizomatica.org>  Tue, 12 Mar 2024 21:01:01 +0000
+
+wiringpi (2.70-4) unstable; urgency=low
```
Comment (mwallner) [inline debian/changelog:7]: please ditch all releases unofficial releases that are not available through wiringpi/wiringpi releases on GitHub

**Review by mwallner — CHANGES_REQUESTED** [review]
please see the review comments

**Comment by rafael2k** [comment]
Just one comment - the branch name is wrong. It should be debian, not ubuntu. I'll take a look in the comments and change the PR as soon as I get some time, thanks!

File: debian/control (hunk 2 above)
Comment (rafael2k) [reply debian/control]: Done.

File: debian/changelog (hunk 1 above)
Comment (rafael2k) [reply debian/changelog]: Changed to stable.

File: debian/copyright
Comment (rafael2k) [reply debian/copyright:1]: Done.

File: debian/changelog (hunk 3 above)
Comment (rafael2k) [reply debian/changelog]: Done.

**Comment by rafael2k** [comment]
Thinking a bit about the "ubuntu" branch. It should go away, and the debian dir should just be in master. What is the rationale of having a separate branch just to hold the debian/ directory, considering all software code will be the same?

**Comment by rafael2k** [comment]
Btw, you talk about "We are aiming for". I'd like to know if this "WiringPi Community" is a closed community, owned by G2, which does not allow new contributors. If it is the case, I'm sorry to bother.

File: debian/control (hunk 2 above)
Comment (rafael2k) [reply debian/control]: Btw, I did not see any link to subscribe or apply to be part of the community.

**Comment by mwallner** [comment]
> Thinking a bit about the "ubuntu" branch. It should go away, and the debian dir should just be in master. What is the rationale of having a separate branch just to hold the debian/ directory, considering all software code will be the same?

it should definitely not go into master, develop would be fine though.
=> 'master' is something people land up on and build their binaries from, I'd like to have master in a mostly stable state,- therefore I'm pushing hard for a develop/master workflow.
active development, new features go into develop, these can be 'released' into master when we're preparing a new release. (which is what we're aiming for doing more frequently, - supported by GitHub actions when I get the time)

**Comment by mwallner** [comment]
> Btw, you talk about "We are aiming for". I'd like to know if this "WiringPi Community" is a closed community, owned by G2, which does not allow new contributors. If it is the case, I'm sorry to bother.

WiringPi is not closed community, - but we (_we_ as GC2) don't have the keys add new people to it (yet, hopefully).

When I currently state 'we' my intended meaning of the word is 'the people actively working on WiringPi and talking about it/ engaging in discussions, issues and what not' - this absolutely includes you.

What _I'm_ trying to do is to be rigor about keeping this project awesome for everyone who's been using it in the past and hopefully for everyone who will stumble upon it in the future.

for the sake of transparency:
Next on _my_ (virtual) list for WiringPi would be getting the packaging up to speed (which includes this issue) as well as reviving some of the wrapper-libs for various other languages than C.

**Comment by rafael2k** [comment]
Thanks for the kind words.
: )
I like the idea of splitting a bit more the package to have a "-bin".
I'll do this, and then at some point I'll send some emails to try to start the process in debian packaging / package maintainers.

**Comment by seamusdemora** [comment]
I have to say, "I am amazed!"  I the past month or two you guys have done more to revive/resuscitate wiringPi than was done during the entire "previous" time. My hat is off to you.

**Pull request merged**